app/models/saved_claim.rb

```diff
@@ -91,8 +91,23 @@ def form_must_be_string
   def form_matches_schema
     return unless form_is_string
 
-    JSON::Validator.fully_validate(VetsJsonSchema::SCHEMAS[self.class::FORM], parsed_form).each do |v|
-      errors.add(:form, v.to_s)
+    schema = VetsJsonSchema::SCHEMAS[self.class::FORM]
+
+    schema_errors = JSON::Validator.fully_validate_schema(schema, { errors_as_objects: true })
+    clear_cache = false
+    unless schema_errors.empty?
+      Rails.logger.error('SavedClaim schema failed validation! Attempting to clear cache.', { errors: schema_errors })
+      clear_cache = true
+    end
+
+    validation_errors = JSON::Validator.fully_validate(schema, parsed_form, { errors_as_objects: true, clear_cache: })
+
+    validation_errors.each do |e|
+      errors.add(e[:fragment], e[:message])
+    end
+
+    unless validation_errors.empty?
+      Rails.logger.warn('SavedClaim form did not pass validation', { guid:, errors: validation_errors })
```

Comment:
What is the reason behind this being a Rails.logger.warn and not an Rails.logger.error?

Reply:
I set it as warn because I believe this can be triggered by anything a user submits that does not match the schema validation, this is probably prevented by front-end validation but it is still possible.